pico/processor/src/main/java/io/helidon/pico/processor/ServiceAnnotationProcessor.java

```diff
@@ -0,0 +1,217 @@
+                          + " @ " + Utils.rootStackTraceElementOf(t), t);
+            // we typically will not even get to this next line since the messager.error() call will trigger things to halt
+            throw new ToolsException("error during processing: " + t
+                                             + " @ " + Utils.rootStackTraceElementOf(t), t);
+        } finally {
+            if (roundEnv.processingOver()) {
+                servicesToProcess().clearModuleName();
+            }
+        }
+    }
+
+    String getThisModuleName(ModuleInfoDescriptor descriptor) {
+        // user pass it in? if so then that will short-circuit the name calculation part
+        String moduleName = Options.getOption(Options.TAG_MODULE_NAME).orElse(null);
+        if (moduleName != null) {
+            return moduleName;
+        }
+        return (descriptor == null) ? null : descriptor.name();
+    }
+
+    ModuleInfoDescriptor getThisModuleDescriptor(AtomicReference<String> typeSuffix,
+                                                 AtomicReference<File> moduleInfoFile,
+                                                 AtomicReference<File> srcPath) {
+        ModuleInfoDescriptor descriptor = tryFindModuleInfo(StandardLocation.SOURCE_OUTPUT,
+                                                              typeSuffix,
+                                                              moduleInfoFile,
+                                                              srcPath);
+        if (descriptor == null) {
+            descriptor = tryFindModuleInfo(StandardLocation.SOURCE_PATH, typeSuffix, moduleInfoFile, srcPath);
+            if (descriptor == null) {
+                // attempt to retrieve from src/main/java if we can't recover to this point
+                descriptor = getThisModuleDescriptorFromSourceMain(moduleInfoFile, srcPath);
+            }
+        }
+        // note to self: this seems worthy to add at some point
+        //        if (Objects.nonNull(srcPath) && Objects.isNull(srcPath.get())) {
+        //            // try to determine the current source path..
+        //            try {
+        //                Filer filer = processingEnv.getFiler();
+        //                for (Element e : roundEnv.getRootElements()) {
+        //                    if (!(e instanceof TypeElement)) {
+        //                        continue;
+        //                    }
+        //                    TypeName typeName = TypeNameImpl.toName(e);
+        //                    FileObject f = filer.getResource(StandardLocation.SOURCE_PATH, typeName.getPackageName
+        //                    (), typeName.getClassName() + ".java");
+        //                    File location = ModuleUtils.toFile(f.toUri());
+        //                }
+        //            } catch (Exception e) {
+        //                debug("unable to determine source path: " + e.getMessage(), null);
+        //            }
+        //        }
```

Comment:
Leftover

Reply:
Suggestion for an alternative?  Just looking for a pass-by-reference option, and couldn't think or find anything better.